File: Backend - SpringBoot/src/main/java/com/estupiso/model/FotoVivienda.java
package com.estupiso.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.validator.constraints.URL;

@Entity
@Data
public class FotoVivienda extends DomainEntity {

    @URL
    private String imagen;

    @ManyToOne(cascade = CascadeType.REMOVE)
    @JoinColumn(name = "vivienda_id", nullable = false)
    @JsonBackReference
    private Vivienda vivienda;

}

File: Frontend - Angular/src/app/model/FotoVivienda.ts
import { Vivienda } from "./Vivienda";

export class FotoVivienda {
    id!: number;
    imagen: string;
    vivienda: Vivienda;

    constructor(imagen: string, vivienda: Vivienda) {
        this.imagen = imagen;
        this.vivienda = vivienda;
    }
}